src/Magnum/Trade/MeshData.h

```diff
@@ -1717,6 +1736,54 @@ class MAGNUM_TRADE_EXPORT MeshData {
          */
         void objectIdsInto(Containers::StridedArrayView1D<UnsignedInt> destination, UnsignedInt id = 0) const;
 
+        /**
+         * @brief Weights as 4D float vectors
+         *
+         * Convenience alternative to @ref attribute(MeshAttribute, UnsignedInt) const
+         * with @ref MeshAttribute::Weights as the first argument. Converts
+         * the weights array from an arbitrary underlying type and returns it
+         * in a newly-allocated array. Expects that the vertex format is *not*
+         * implementation-specific, in that case you can only access the
+         * attribute via the typeless @ref attribute(MeshAttribute, UnsignedInt) const.
+         * @see @ref weightsInto(), @ref attributeFormat(),
+         *      @ref isVertexFormatImplementationSpecific()
+         */
+        Containers::Array<Vector4> weightsAsArray(UnsignedInt id = 0) const;
+
+        /**
+         * @brief Weights as 4D float vectors into a pre-allocated view
+         *
+         * Like @ref weightsAsArray(), but puts the result into
+         * @p destination instead of allocating a new array. Expects that
+         * @p destination is sized to contain exactly all data.
+         * @see @ref vertexCount()
+         */
+        void weightsInto(Containers::StridedArrayView1D<Vector4> destination, UnsignedInt id = 0) const;
+
+        /**
+         * @brief Weights as 4D unsigned short vectors
+         *
+         * Convenience alternative to @ref attribute(MeshAttribute, UnsignedInt) const
+         * with @ref MeshAttribute::JointIndices as the first argument. Converts
+         * the joint indices array from an arbitrary underlying type and returns it
+         * in a newly-allocated array. Expects that the vertex format is *not*
+         * implementation-specific, in that case you can only access the
+         * attribute via the typeless @ref attribute(MeshAttribute, UnsignedInt) const.
+         * @see @ref jointIndicesInto(), @ref attributeFormat(),
+         *      @ref isVertexFormatImplementationSpecific()
+         */
+        Containers::Array<Vector4us> jointIndicesAsArray(UnsignedInt id = 0) const;
```

Comment:
I think should always be the largest type (even if it will never be as big):

```suggestion
        Containers::Array<Vector4ui> jointIndicesAsArray(UnsignedInt id = 0) const;
```